src/transformers/modeling_utils.py

```diff
@@ -1649,6 +1652,13 @@ def _check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> Pretra
 
         if not hard_check_only:
             config._attn_implementation = "sdpa"
+
+        if torch.version.hip is not None and config._attn_implementation == "sdpa" and device_map == "auto":
+            logger.warning_once(
+                "Using the `SDPA` attention implementation with `device_map='auto'` on a ROCM device may lead to performance issues due to the FA backend. Disabling it to use alternative backends."
```

Comment:
Did not have time for this, but ideally we should have a pytorch issue open with a repro without transformers, and link it here

Reply:
I agree. But not sure if it is also bind to `auto` in `transformers`.

Reply:
Also, it might be good to put all of these into a AMD documentation, so we can share with them.